src/main/groovy/com/lesfurets/jenkins/unit/BasePipelineTest.groovy

```diff
@@ -149,6 +149,24 @@ abstract class BasePipelineTest {
                 updateBuildStatus('FAILURE')
             }
         }
+        helper.registerAllowedMethod('catchError', [Map,Closure]) { Map args, Closure c ->
+            c.delegate = delegate
+            try {
+                helper.callClosure(c)
+            } catch(ignored) {
+                if(args.message) {
+                    println args.message
+                }
+                if ( args.buildResult == null ) {
+                    return
+                }
+                if ('SUCCESS' == binding.getVariable('currentBuild').result) {
+                    updateBuildStatus(args.buildResult as String)
+                } else if ('UNSTABLE' == binding.getVariable('currentBuild').result && 'FAILURE' == args.buildResult ) {
+                    updateBuildStatus('FAILURE')
+                }
```

Comment:
IMHO, this is a clearer interpretation of the [documentation](https://www.jenkins.io/doc/pipeline/steps/workflow-basic-steps/#catcherror-catch-error-and-set-build-result-to-failure):

```suggestion
                if (binding.getVariable('currentBuild').currentResult == 'FAILURE') {
                    // The build result can only get worse, so we must ignore the desired buildResult argument
                    return
                }
                updateBuildStatus(args.buildResult)
```